xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/pdfheader.vm

```diff
@@ -1,6 +1,10 @@
 #set ($title = "$!{pdfdoc.display('title', 'rendered', '', $pdfobj)}")
 #if ($title == '')
-  $!{escapetool.xml($doc.space)} - $!{escapetool.xml($tdoc.plainTitle)}
+  #if ($doc.space != $tdoc.plainTitle)
+    $!{escapetool.xml($doc.space)} - $!{escapetool.xml($tdoc.plainTitle)}
+  #else
+    $!{escapetool.xml($tdoc.plainTitle)}
```

Comment:
A small improvement could also be to display the full reference, see XWIKI-14260